src/bin/bors.rs

```diff
@@ -0,0 +1,63 @@
+use std::net::SocketAddr;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use anyhow::Context;
+use axum::http::StatusCode;
+use axum::response::IntoResponse;
+use axum::routing::post;
+use axum::Router;
+use clap::Parser;
+use tower::limit::ConcurrencyLimitLayer;
+
+use bors::config::Config;
+use bors::github::webhook::GitHubWebhook;
+
+#[derive(clap::Parser)]
+struct Opts {
+    /// Path to a config file.
+    config_file: PathBuf,
+}
+
+async fn github_webhook(GitHubWebhook(event): GitHubWebhook) -> impl IntoResponse {
+    println!("Received GitHub webhook: {:?}", event);
+    (StatusCode::OK, "")
+}
+
+async fn server(config: Config) -> anyhow::Result<()> {
+    let state = Arc::new(config);
+
+    let app = Router::new()
+        .route("/github", post(github_webhook))
+        .layer(ConcurrencyLimitLayer::new(100))
+        .with_state(state);
+    let addr = SocketAddr::from(([0, 0, 0, 0], 8080));
+
+    axum::Server::bind(&addr)
+        .serve(app.into_make_service())
+        .await?;
+    Ok(())
+}
+
+fn try_main(opts: Opts) -> anyhow::Result<()> {
+    let config = Config::from_toml_file(&opts.config_file).context("Cannot load config file")?;
+
+    let runtime = tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .context("Cannot build tokio runtime")?;
+
+    let server_fut = runtime.spawn(server(config));
+    runtime.block_on(server_fut)??;
+
+    Ok(())
+}
+
+fn main() {
+    env_logger::init();
+
+    let opts = Opts::parse();
+    if let Err(error) = try_main(opts) {
+        eprintln!("Error: {error:?}");
```

Comment:
This returns with exit code 0 if there is an error.

Reply:
Fixed, thanks.